src/main/java/org/datadog/jmxfetch/JmxAttribute.java

```diff
@@ -136,9 +137,12 @@ private void addAdditionalTags() {
         }
     }
 
-    private void addServiceTag() {
-        if (serviceName != null && !serviceName.isEmpty()) {
-            this.defaultTagsList.add("service:" + serviceName);
+    private void addServiceTags() {
+        Iterable<String> serviceNames = this.serviceNameProvider.getServiceNames();
+        if (serviceNames != null) {
+            for (String serviceName : serviceNames) {
+                this.defaultTagsList.add("service:" + serviceName);
```

Comment:
Does order matter with these calls? ie: do we need to return a stably ordered collection?

Reply:
Order does not matter, we should be good.